.github/workflows/tests.yml

```diff
@@ -0,0 +1,30 @@
+name: Tests
+on:
+  pull_request:
+    branches:
+      - main
+      - support/**
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [ '10.x', '12.x' ]
+    name: Node ${{ matrix.node }}
+    steps:
+    - uses: actions/checkout@v2
+    - run: |
+        git remote set-branches --add origin main
```

Comment:
I've seen other `one` projects do this, but I have not seen this on the checkout action for main branches. Does an error occur if it's missing?

Reply:
I haven't tested without it as I stole this directly from @nellyk. @nellyk, is `actions/checkout@v2` doing a shallow clone here?

Reply:
If there was an issue with the default branch not being `master` before, I believe it was resolved in actions/checkout#305 and the above is no longer necessary.

Re: shallow clones `actions/checkout` will only grab a single commit unless you provide `fetch-depth: 0` as a parameter.